intellij/src/saros/intellij/ui/wizards/AddProjectToSessionWizard.java

```diff
@@ -310,13 +282,70 @@ private void doExistingModule(@NotNull Module existingModule) {
             return;
           }
 
+          Map<String, String> moduleParameters =
+              negotiation.getProjectNegotiationData(remoteProjectID).getAdditionalProjectData();
+
+          ModuleConfiguration moduleConfiguration = new ModuleConfiguration(moduleParameters, true);
+
+          /*
+           * TODO move to finish of FilesChangedPage iff back button is configured to function as
+           *  the module might change in that case
+           */
+          queueModuleConfigurationChange(existingModule, moduleConfiguration);
+
           localProjects.put(remoteProjectID, sharedProject);
 
           prepareFilesChangedPage(localProjects);
 
           setTopPanelText(Messages.AddProjectToSessionWizard_description_changed_files);
         }
 
+        /**
+         * Adds the given module configuration to the {@link ModuleConfigurationInitializer}.
+         *
+         * <p>Cancels the negotiation if the session or session context is no longer valid.
+         *
+         * @param module the module the configuration belongs to
+         * @param moduleConfiguration the module configuration
+         */
+        private void queueModuleConfigurationChange(
+            @NotNull Module module, @NotNull ModuleConfiguration moduleConfiguration) {
+
+          ISarosSession session = sessionManager.getSession();
+
+          if (session == null) {
```

Comment:
The session manager already calls cancel. And so the cancelListener should be notified already.

Reply:
Done. Removed call to `cancelNegotiation(String)`.

Reply:
Looking at the code, when the wizard is cancelled a popup is already displayed. Of course it does not hurt however not displaying anything would  not hurt either.